aot/src/main/java/com/dylibso/chicory/aot/AotEmitters.java

```diff
@@ -95,35 +93,74 @@ public static Builder builder() {
         return new Builder();
     }
 
-    public static void DROP(AotContext ctx, AnnotatedInstruction ins, MethodVisitor asm) {
-        emitPop(asm, ctx.popStackSize());
+    public static void TRAP(AotContext ctx, AotInstruction ins, MethodVisitor asm) {
+        emitInvokeStatic(asm, THROW_TRAP_EXCEPTION);
+        asm.visitInsn(Opcodes.ATHROW);
+    }
+
+    public static void DROP_KEEP(AotContext ctx, AotInstruction ins, MethodVisitor asm) {
+        int keepStart = (int) ins.operand(0) + 1;
+
+        // save result values
+        int slot = ctx.tempSlot();
+        for (int i = ins.operandCount() - 1; i >= keepStart; i--) {
+            var type = ValueType.forId((int) ins.operand(i));
+            asm.visitVarInsn(storeTypeOpcode(type), slot);
+            slot += slotCount(type);
+        }
+
+        // drop intervening values
+        for (int i = keepStart - 1; i >= 1; i--) {
```

Comment:
is this a stack `unwind`?

Reply:
Yes, exactly, this is why I've been thinking to name the opcode `UNWIND` instead.